gslab_scons/tests/test_configuration_tests.py

```diff
@@ -0,0 +1,394 @@
+import unittest
+import sys
+import os
+import re
+import shutil
+import mock
+import datetime
+import subprocess
+
+# Ensure that Python can find and load the GSLab libraries
+os.chdir(os.path.dirname(os.path.realpath(__file__)))
+sys.path.append('../..')
+
+import gslab_scons.configuration_tests as configuration_tests
+import gslab_scons._exception_classes as ex_classes
+from gslab_make.tests import nostderrout
+
+
+class TestConfigurationTests(unittest.TestCase):
+
+    @mock.patch('gslab_scons.configuration_tests.check_python_packages')
+    def test_check_python(self, mock_check_python_packages):
+        # Incorrect python
+        with mock.patch('gslab_scons.configuration_tests.sys.version_info', [3]):
+            with self.assertRaises(ex_classes.PrerequisiteError):
+                configuration_tests.check_python('3.0.2')
+            mock_check_python_packages.assert_not_called()
+
+        with mock.patch('gslab_scons.configuration_tests.sys.version_info', [2]):
+            # Test default arguments and correct python
+            configuration_tests.check_python('3.0.2')
+            mock_check_python_packages.assert_called_with('3.0.2', ["yaml", "gslab_scons", 
+                                                           "gslab_make", "gslab_fill"])
+
+            # Test alternative arguments and correct python
+            packages = ["yaml"]
+            configuration_tests.check_python(3.0, packages)
+            mock_check_python_packages.assert_called_with(3.0, ["yaml"])        
+
+            # Test alternative arguments format and correct python
+            packages = "yaml"
+            configuration_tests.check_python('3.0.2', packages)
+            mock_check_python_packages.assert_called_with('3.0.2', "yaml") 
+
+
+    @mock.patch("gslab_scons.configuration_tests.pkg_resources.get_distribution")
+    @mock.patch('gslab_scons.configuration_tests.importlib.import_module')
+    @mock.patch('gslab_scons.configuration_tests.convert_packages_argument')
+    def test_check_python_packages(self, mock_convert_packages, mock_import, mock_version):
+        # Setup
+        class MockVersionSideEffect(object):
+            def __init__(self):
+                self.version = '3.0.2'
+
+        def version_side_effect(*args, **kwargs):
+            return MockVersionSideEffect()
+
+        mock_version.side_effect          = version_side_effect
+        mock_convert_packages.side_effect = lambda x: ['yaml', 'os']
+        mock_import.side_effect           = lambda x: None
+
+        # Default correct test
+        configuration_tests.check_python_packages('3.0.2', ['yaml', 'os'])
+        mock_convert_packages.assert_called_with(['yaml', 'os'])
+        mock_import.assert_called()
+        mock_version.assert_called_with('gslab_python')
+
+        # Incorrect gslab_python version
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            configuration_tests.check_python_packages('3.0.3', ['yaml', 'os'])
```

Comment:
Maybe we could also have a test that ensure the lexicographical ordering issue I mentioned [here]() doesn't cause unintended behaviour?